4-WebApp-your-API/4-1-MyOrg/TodoListService/Controllers/TodoListController.cs

```diff
@@ -27,66 +31,161 @@ public TodoListController(IHttpContextAccessor contextAccessor)
             // Pre-populate with sample data
             if (TodoStore.Count == 0)
             {
-                TodoStore.Add(1, new Todo() { Id = 1, Owner = $"{_contextAccessor.HttpContext.User.Identity.Name}", Title = "Pick up groceries" });
-                TodoStore.Add(2, new Todo() { Id = 2, Owner = $"{_contextAccessor.HttpContext.User.Identity.Name}", Title = "Finish invoice report" });
+                TodoStore.Add(1, new Todo() { Id = 1, Owner = $"{GetObjectIdClaim(_contextAccessor.HttpContext.User)}", Title = "Pick up groceries" });
+                TodoStore.Add(2, new Todo() { Id = 2, Owner = $"{GetObjectIdClaim(_contextAccessor.HttpContext.User)}", Title = "Finish invoice report" });
+                TodoStore.Add(3, new Todo() { Id = 3, Owner = "Other User", Title = "Rent a car" });
+                TodoStore.Add(4, new Todo() { Id = 4, Owner = "Other User", Title = "Get vaccinated" });
             }
         }
 
         // GET: api/values
         [HttpGet]
-        [RequiredScope(new string[] { "ToDoList.Read", "ToDoList.Write" })]
+        [RequiredScopeOrAppPermission(
+            AcceptedScope = new string[] { "ToDoList.Read", "ToDoList.ReadWrite" },
+            AcceptedAppPermission = new string[] { "ToDoList.Read.All", "ToDoList.ReadWrite.All" }
+            )]
         public IEnumerable<Todo> Get()
         {
-            string owner = User.Identity.Name;
-            return TodoStore.Values.Where(x => x.Owner == owner);
+            if (HasDelegatedPermissions(new string[] { "ToDoList.Read", "ToDoList.ReadWrite" }))
```

Comment:
It's a get. Do you need to verify the "TodoList.ReadWrite" ?

Reply:
We do it for a case where a user have only ReadWrite scope.